src/pgas.jl

```diff
@@ -0,0 +1,201 @@
+""" 
+    initialization(model::AbstractStateSpaceModel)
+
+Define the distribution of the initial state of the State Space Model
+"""
+function initialization end
+
+""" 
+    transition(model::AbstractStateSpaceModel)
+
+Define the transition density of the State Space Model
+"""
+function transition end
+
+"""
+    observation(model::AbstractStateSpaceModel)
+
+Return the log-likelihood of the observed measurement conditional on the current state of the model. 
+Must return `nothing` if it consumed all the data
+"""
+function observation end
+
+"""
+    inc_step!(model::AbstractStateSpaceModel, n::Int=1)
+
+Increase the internal model counter by `n`. Used to keep track of the current timestep when replaying
+"""
+inc_step!(m::AbstractStateSpaceModel, n::Int=1) = m.T += n
```

Comment:
So every `AbstractStateSpaceModel` is supposed to have a field `T` (BTW maybe use a more descriptive name)? This seems quite specific, maybe better to define this only for the concrete model type that has such a field?

Reply:
This applies to all definitions with `.T` below

Reply:
If we want to plug this into DifferenceEquations.jl, we might use `tspan` instead or otherwise include some kind of subsetting convenience function `subset(prob::StatespaceModel, (lower_t, upper_t))`.

Reply:
I'll change the name, but we need a way to keep track of the "time" parameter somehow. @cpfiffer not too familiar with DifferenceEquations.jl, what's the convention there ?

Reply:
Hm, we don't have one yet. Let's stick with what you have here and we'll shoehorn it in later.